src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+        }
+        return c;
+    }
+
+    /**
+     * <p>
+     * Converts the 4 lsb of an int to a hexadecimal digit encoded using the Msb0 bit ordering.
+     * </p>
+     * <p>
+     * 0 returns '0'
+     * </p>
+     * <p>
+     * 1 returns '8'
+     * </p>
+     * <p>
+     * 10 returns '5' and so on...
+     * </p>
+     *
+     * @param nibble the 4 bits to convert
+     * @return a hexadecimal digit representing the 4 lsb of {@code nibble}
+     * @throws IllegalArgumentException if {@code nibble < 0} or {@code nibble > 15}
+     */
+    public static char intToHexDigitMsb0(final int nibble) {
+        switch (nibble) {
+        case 0x0:
+            return '0';
+        case 0x1:
+            return '8';
+        case 0x2:
+            return '4';
+        case 0x3:
+            return 'c';
+        case 0x4:
+            return '2';
+        case 0x5:
+            return 'a';
+        case 0x6:
+            return '6';
+        case 0x7:
+            return 'e';
+        case 0x8:
+            return '1';
+        case 0x9:
+            return '9';
+        case 0xA:
+            return '5';
+        case 0xB:
+            return 'd';
+        case 0xC:
+            return '3';
+        case 0xD:
+            return 'b';
+        case 0xE:
+            return '7';
+        case 0xF:
+            return 'f';
+        default:
+            throw new IllegalArgumentException("nibble value not between 0 and 15: " + nibble);
+        }
+    }
+
+    /**
+     * <p>
+     * Converts an array of int into a long using the default (little endian, Lsb0) byte and bit
+     * ordering.
+     * </p>
+     *
+     * @param src the int array to convert
+     * @param srcPos the position in {@code src}, in int unit, from where to start the
+     *            conversion
+     * @param dstInit initial value of the destination long
+     * @param dstPos the position of the lsb, in bits, in the result long
+     * @param nInts the number of ints to convert
+     * @return a long containing the selected bits
+     * @throws IllegalArgumentException if {@code (nInts-1)*32+dstPos >= 64}
+     * @throws NullPointerException if {@code src} is {@code null}
+     * @throws ArrayIndexOutOfBoundsException if {@code srcPos + nInts > src.length}
+     */
+    public static long intArrayToLong(final int[] src, final int srcPos, final long dstInit, final int dstPos,
+            final int nInts) {
+        if (src.length == 0 && srcPos == 0 || 0 == nInts) {
+            return dstInit;
+        }
+        if ((nInts - 1) * 32 + dstPos >= 64) {
+            throw new IllegalArgumentException("(nInts-1)*32+dstPos is greater or equal to than 64");
+        }
+        long out = dstInit;
+        for (int i = 0; i < nInts; i++) {
+            final int shift = i * 32 + dstPos;
+            final long bits = (0xffffffffL & src[i + srcPos]) << shift;
+            final long mask = 0xffffffffL << shift;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).